soccer/gameplay/tactics/defense.py

```diff
@@ -403,6 +413,10 @@ def recalculate_threat_shot(threat_index):
         # print("Unused handlers: " + str(unused_threat_handlers))
         # print("---------------------")
 
+        # If we have nothing to block, bail
+        if not threats:
+            return
+
         smart = False
         if not smart:
```

Comment:
The last commit in the smart block was from 3 years ago, and hasn't been touched since. I don't think it's better than the 'non-smart' code in any way.

It also seems to avoid blocking anything but the ball until the ball starts moving towards enemies.

![2017-08-22-114116_1333x1058_scrot](https://user-images.githubusercontent.com/4349709/29574189-bb1e8778-8715-11e7-93f6-0613fdadebe7.png)


Since it's been dead code for that long, I suspect it's completely broken.